apps/labs/posts/polars-plugins-lets-make-them-easier-to-use.md

```diff
@@ -0,0 +1,218 @@
+It didn't take long for my end-goal to be presented to me, which was: **to improve and make it easier to follow a Polars Plugin tutorial** (the one authored by Marco).
+
+I was having daily meetings with him, and he was encouraging me to follow the tutorial as well as explore plugins in general.
+And so I did. I went through the entire tutorial in the first days of the week, and eventually decided to share with him the crazy idea I had before it all started - making Conway's Game of Life "run" in a dataframe.
+What would you tell a lunatic intern with such a ridiculous idea? I don't know, but he said something along the lines of "I'm totally on board with it".
+This was **it**, _the game was on!_ But wait, no... I don't know how to do that... I actually have no clue!
+
+Those were my thoughts, initially. I believed the same limitations I had before the internship began were also present here, but that was not entirely true.
+After all, I did the tutorial, I played with some examples, and I even managed to "explain" to ChatGPT a broader context of plugins, allowing it to give me some useful suggestions besides its usual hallucinations.
+I could at least _try_ something. Fast forward... a day? Two? It was done - for real. I couldn't believe myself.
+
+You see, Conway's Game of Life is a cellular automaton which beautifully displays emergent behavior.
+Complex constructs from a _very simple_ rule-set.
+
+<figure style={{ textAlign: 'center' }}>
+    <img 
+      src="/posts/polars-plugins-lets-make-them-easier-to-use/life_toad.gif"
+      alt="GIF of a terminal displaying a pattern formed by filled or empty cells alternating between two states"
+      style={{ display: 'inline-block' }}
+    />
+    <figcaption>Toad pattern</figcaption> 
+</figure>
+
+What needed to be done? I had to get the rule-set right, and manage to advance the simulation (iterate the rules).
+Assuming the dataframe would contain 0's and 1's only, I went ahead and broke the challenge into two smaller groups:
+
+1. I had to implement a plugin in Rust to sum the values of both neighbouring rows and columns. Then I had to decide what the element in the center would be in the next iteration: dead or alive, 0 or 1.
+2. I needed to find a way to step the simulation. For that, I wrote a Python script to repeatedly call the plugin, preserving the names of the columns (this way, columns from a new iteration would replace columns from the older one - this is a Polars specific detail).
+
+After finishing those, the magic took place, the simulation was happening before my eyes:
+
+<figure style={{ textAlign: 'center' }}>
+    <img 
+      src="/posts/polars-plugins-lets-make-them-easier-to-use/life_toad_df.gif"
+      alt="GIF of a dataframe displaying a pattern formed by 1's and 0's alternating between two states"
+      style={{ display: 'inline-block' }}
+    />
+    <figcaption>Toad pattern, in a dataframe</figcaption> 
```

Comment:
Maybe add a link to what a Toad pattern is?

Reply:
Then I suggest either including a link to a wikipedia page or some standard reference explaining what it it, or just not naming it at all (e.g. "Conway's game of life pattern, shown in a dataframe")

Reply:
Removed the name of the pattern to avoid confusion, thanks!